### PR TITLE
Add pip check, some build_number math

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

File: .azure-pipelines/azure-pipelines-linux.yml [hunk 1]
```diff
@@ -8,76 +8,16 @@ jobs:
...
-        DOCKER_IMAGE: quay.io/condaforge/linux-anvil-cos7-x86_64
-      linux_aarch64_python3.9.____73_pypy:
-        CONFIG: linux_aarch64_python3.9.____73_pypy
-        UPLOAD_PACKAGES: 'True'
-        DOCKER_IMAGE: quay.io/condaforge/linux-anvil-cos7-x86_64
-      linux_aarch64_python3.9.____cpython:
-        CONFIG: linux_aarch64_python3.9.____cpython
-        UPLOAD_PACKAGES: 'True'
-        DOCKER_IMAGE: quay.io/condaforge/linux-anvil-cos7-x86_64
-      linux_ppc64le_python3.10.____cpython:
```
Comment (Marco-Sulla) [inline .azure-pipelines/azure-pipelines-linux.yml:59]: ? Why you removed ppc64?

File: .azure-pipelines/azure-pipelines-osx.yml [hunk 2]
```diff
@@ -8,38 +8,11 @@ jobs:
...
-      osx_64_python3.8.____cpython:
-        CONFIG: osx_64_python3.8.____cpython
-        UPLOAD_PACKAGES: 'True'
-      osx_64_python3.9.____73_pypy:
-        CONFIG: osx_64_python3.9.____73_pypy
-        UPLOAD_PACKAGES: 'True'
-      osx_64_python3.9.____cpython:
-        CONFIG: osx_64_python3.9.____cpython
-        UPLOAD_PACKAGES: 'True'
-      osx_arm64_python3.10.____cpython:
```
Comment (Marco-Sulla) [inline .azure-pipelines/azure-pipelines-osx.yml:29]: Why you removed arm?

**Review by Marco-Sulla — COMMENTED** [review]
I do not understand the scope of this merge request.

File: .azure-pipelines/azure-pipelines-linux.yml (hunk 1 above)
Comment (beckermr) [reply .azure-pipelines/azure-pipelines-linux.yml:59]: These changes have not removed these platforms at all. See the new set of ci support files in the PR.

File: .azure-pipelines/azure-pipelines-osx.yml (hunk 2 above)
Comment (beckermr) [reply .azure-pipelines/azure-pipelines-osx.yml:29]: These changes have not removed these platforms at all. See the new set of ci support files in the PR.

**Comment by erykoff** [comment]
I don't know if I should comment on this PR or (probably) make a new issue, but the current meta.yaml seems to be giving incorrect outputs.  You can't mix `noarch` and arch with the same output name, and noarch can't have python constraints.  Now the frozendict 2.3.10 installation seems to always give noarch, and built with 3.12 so it doesn't work right with 3.11.  I think we'd need a different output name?  (Not sure, this is beyond my depth except to say that we're not getting the right file)

**Comment by bollwyvl** [comment]
Yep, sorry, was just trying some things. Basically, downstream, I'm getting:
- the `noarch` package when there exists a (presumably better) platform/python-specific version
- some shouty DEPRECATED warnings during `pip check` about the `.egg` that ends up from the `setup.py install`

Perhaps a re-render on `main` would decrease the noise.

**Comment by beckermr** [comment]
You can build noarch on a more recent python than what you run on. Something else happened which I cannot quite figure out.

**Comment by beckermr** [comment]
@Marco-Sulla Why does the noarch build contain a python .egg and not a straight copy of the source like I am more familiar with?

**Comment by beckermr** [comment]
Tbh I don't think having noarch and arch builds for the same package makes sense. Noarch will always be picked since it comes really high in the solver priority.

**Comment by bollwyvl** [comment]
Welp, in the interim, I've worked around depending on this package by paring down some other deps, so will close this... good luck!

**Comment by Marco-Sulla** [comment]
> Tbh I don't think having noarch and arch builds for the same package makes sense. Noarch will always be picked since it comes really high in the solver priority.

This is not how `pip` works. I do not really understand the sense of giving higher priority to noarch.

**Comment by beckermr** [comment]
Yeah conda is not pip. This preference reduces the sizes of conda installs, but in general IDK why it is that way.

**Pull request closed without merging**